src/services/modules/rest/impl/src/main/java/org/geoserver/geofence/services/rest/impl/RESTRuleServiceImpl.java

```diff
@@ -88,7 +93,6 @@ public RESTOutputRule get(Long id) throws BadRequestRestEx, NotFoundRestEx, Inte
     }
 
     @Override
-    @Transactional(propagation = Propagation.REQUIRED, value = "geofenceTransactionManager")
```

Comment:
I would not remove the Transactional annotation here: there are many db calls and race conditions may disrupt the priority handling.

Reply:
fixed